### ISSUE TITLE
Isolate your excerpts and don't interfere with theme / PR for Elemetor Pro skin-base.php

### ISSUE THREAD

**Issue by carasmo** [elementor/elementor#5620]
## Prerequisites

- [ x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [ x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**

Many themes have modified the excerpt_more and also the excerpt_length. You should isolate your content grabbing so that filters that exist don't have to be removed on Elementor pro query output. So instead of using the excerpt_more filter, which as you can see in the image, I've already modified for most of the site, when Elementor gets the content, it should be the raw, unfiltered content, then you make your own Elementor filters so that it doesn't touch the theme's filters. 

I want to add a `&hellip;` to the end of the excerpt in Elementor's output.

**Describe the solution you'd like**

When getting the content for a query, don't use the filters inside WordPress, make your own get the raw content and build it so that it's isolated. This is how Generate press does it, and how it should be done.

Allow ... `&hellip;` on the excerpt. 

**HELP**
Need to be able to get the ellipsis after the excerpt like this:

**Title**
Excerpt words go here...

_Read more link_

**Describe alternatives you've considered**
Not using Elementor for loops. Build it then create a shortcode and stick it in Elementor. 

**Add a filter and get raw get_the_excerpt. For skin-base.php**

_This isolates from the theme's filters on the_excerpt and adds a filter 'elementor_excerpt_end' to add the `&nbsp;&hellip;`_

	protected function render_excerpt() {
		add_filter( 'excerpt_more', [ $this, 'filter_excerpt_more' ], 20 );
		add_filter( 'excerpt_length', [ $this, 'filter_excerpt_length' ], 20 );

		if ( ! $this->get_instance_value( 'show_excerpt' ) ) {
			return;
		}

		add_filter( 'excerpt_more', [ $this, 'filter_excerpt_more' ], 20 );
		add_filter( 'excerpt_length', [ $this, 'filter_excerpt_length' ], 20 );
		
		?>
		<div class="elementor-post__excerpt">
		
			<?php 
			
			$excerpt = get_the_excerpt();
						
			$excerpt = apply_filters( 'elementor_excerpt_end', $excerpt . '' );
			
			echo wpautop( $excerpt );
			
			?>
			
		</div>
		<?php

		remove_filter( 'excerpt_length', [ $this, 'filter_excerpt_length' ], 20 );
		remove_filter( 'excerpt_more', [ $this, 'filter_excerpt_more' ], 20 );
		
	}

<img width="1371" alt="screen shot 2018-08-30 at 8 13 24 am" src="https://user-images.githubusercontent.com/1124594/44851162-6bfc4200-ac2d-11e8-9259-e499a5ec183b.png">

**In the meantime:**
I've removed my filter for the read more on Elementor Pro output. Next got to figure out the ...

```
add_action( 'elementor/element/before_section_start', function() {
	remove_filter( 'the_excerpt', 'christina_excerpt_read_more_link' );
} );

```

**Comment by carasmo** [comment]
Well, I can't figure out how to add the ellipses at the end of the excerpt within Elementor, except with JS.

**Comment by carasmo** [comment]
This is what I propose for skin-base.php for the render_excerpt function. A filter:

	protected function render_excerpt() {
		add_filter( 'excerpt_more', [ $this, 'filter_excerpt_more' ], 20 );
		add_filter( 'excerpt_length', [ $this, 'filter_excerpt_length' ], 20 );

		if ( ! $this->get_instance_value( 'show_excerpt' ) ) {
			return;
		}

		add_filter( 'excerpt_more', [ $this, 'filter_excerpt_more' ], 20 );
		add_filter( 'excerpt_length', [ $this, 'filter_excerpt_length' ], 20 );
		
		?>
		<div class="elementor-post__excerpt">
		
			<?php 
			
			$excerpt = get_the_excerpt();
						
			$excerpt = apply_filters( 'elementor_excerpt_end', $excerpt . '' );
			
			echo wpautop( $excerpt );
			
			?>
			
		</div>
		<?php

		remove_filter( 'excerpt_length', [ $this, 'filter_excerpt_length' ], 20 );
		remove_filter( 'excerpt_more', [ $this, 'filter_excerpt_more' ], 20 );
		
	}

**Comment by Aviad-Herman** [comment]
Hi,
I'm closing this old issue.
Feel free to re-open if it's still relevant

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor Pro v3.13.0! 🥳 

You can now use the Loop Builder to add the Post Excerpt widget and include the ellipsis at the end of the Post Excerpt.

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
